### ISSUE TITLE
[Bug] [Android] MissingMethodException: No constructor found for Xamarin.CommunityToolkit.Android.Effects.PlatformTouchEffect+AccessibilityListener::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership)

### ISSUE THREAD

**Issue by thisisthekap** [xamarin/XamarinCommunityToolkit#1923]
### Description

This happens sometimes in production when the app was sent to background or resumes from background. I suppose that the system tries to restore an instance of AccessibilityListener using the specified constructor. This may be fixable by adding this constructor while doing the appropriate base call.

### Stack Trace

```
System.NotSupportedException: Unable to activate instance of type Xamarin.CommunityToolkit.Android.Effects.PlatformTouchEffect+AccessibilityListener from native handle 0x7fc8d60df4 (key_handle 0xd8e584).
  at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x0017e] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x00023] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  at Java.Lang.Object.GetObject[T] (System.IntPtr jnienv, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00006] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  at Android.Views.Accessibility.AccessibilityManager+IAccessibilityStateChangeListenerInvoker.n_OnAccessibilityStateChanged_Z (System.IntPtr jnienv, System.IntPtr native__this, System.Boolean enabled) [0x00000] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.103(intptr,intptr,bool)
--- End of inner exception stack trace ---
  System.MissingMethodException: No constructor found for Xamarin.CommunityToolkit.Android.Effects.PlatformTouchEffect+AccessibilityListener::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership)
    at Java.Interop.TypeManager.CreateProxy (System.Type type, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x000b5] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
    at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00111] in <0c41f4e7d26b446fb63d4ef4709d3b54>:0
  --- End of inner exception stack trace ---
    Java.Interop.JavaLocationException: Exception of type 'Java.Interop.JavaLocationException' was thrown.
      at Java.Lang.Error: Exception of type 'Java.Lang.Error' was thrown.
      at java.lang.Error: Java callstack:
      at crc64515ee83f00766c60.PlatformTouchEffect_AccessibilityListener.n_onAccessibilityStateChanged(Native Method)
      at crc64515ee83f00766c60.PlatformTouchEffect_AccessibilityListener.onAccessibilityStateChanged(Unknown Source:0)
      at android.view.accessibility.AccessibilityManager.lambda$notifyAccessibilityStateChanged$0(AccessibilityManager.java:1305)
      at android.view.accessibility.-$$Lambda$AccessibilityManager$yzw5NYY7_MfAQ9gLy3mVllchaXo.run(Unknown Source:4)
      at android.os.Handler.handleCallback(Handler.java:883)
      at android.os.Handler.dispatchMessage(Handler.java:100)
      at android.os.Looper.loop(Looper.java:221)
      at android.app.ActivityThread.main(ActivityThread.java:7542)
      at java.lang.reflect.Method.invoke(Native Method)
      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

### Link to Reproduction Sample

Unfortunately not reproducible.

### Steps to Reproduce

Unfortunately not reproducible.

### Expected Behavior

No crash.

### Actual Behavior

Crash.

### Basic Information

- Version with issue: 1.3.0
- Last known good version: -
- IDE: Jetbrains Rider
- Platform Target Frameworks:
  - Android: 12.0 
- Affected Devices: Various Samsung and Redmi devices.

### Workaround

None.

### Reproduction imagery

Unfortunately not reproducible.

**Comment by pictos** [comment]
@thisisthekap can you please create a repro project and share the github's link with us?

**Comment by thisisthekap** [comment]
@pictos Unfortunately, the issue is not reproducible.

**Comment by pictos** [comment]
But in the description, you mentioned when the issue happens, and how to reproduce it 🤔

**Comment by thisisthekap** [comment]
As stated it happened in production. Our reporting gave us the data to figure out the user actions that were made.

**Comment by thisisthekap** [comment]
@pictos Any suggestion on how to proceed with this issue and the related PR?

**Comment by pictos** [comment]
@thisisthekap I'll review your PR on my end ASAP, then it will be approved and merged

**Comment by thisisthekap** [comment]
@pictos Awesome! Thank you very much!

**Comment by thisisthekap** [comment]
Hi @pictos!

Are you able to provide a rough estimate when my PR is going to be reviewed?
